Features/cgu/24.2/resize move 389402

cguglielmo opened the pull request.

cguglielmo committed:
Resizable: wrong cursor during resizing

The resize cursor should be visible during the whole resize process.
Currently, it is only visible when the cursor is on the resizable
handle.

Also simplified Resizable.ts a little bit and made sure resize overlay
is removed when resizable is destroyed.

389402

cguglielmo committed:
TileResizeHandler: make resizing more reactive

The resizing now reacts on the gaps between the cells as well.

389402

cguglielmo committed:
Resizable/MoveSupport: cancel resizing when pressing ESC

389402

bschwarzent commented on eclipse-scout-core/src/form/fields/smartfield/TreeProposalChooser.ts:
```diff
@@ -215,6 +215,18 @@ export class TreeProposalChooser<TValue> extends ProposalChooser<TValue, Tree, P
   }
 
+  /**
+   * This function creates a list of flat tree nodes from a list of lookup rows.
```
```suggestion
   * This function creates a flat list of tree nodes from a list of lookup rows.
```

cguglielmo replied:
This commit should not be in this PR.... :/

bschwarzent commented on eclipse-scout-core/src/form/fields/smartfield/TreeProposalChooser.ts:
```diff
@@ -83,7 +83,7 @@ export class TreeProposalChooser<TValue> extends ProposalChooser<TValue, Tree, P
       appendResult = scout.nvl(result.appendResult, false);
 
     if (appendResult) {
-      treeNodesFlat = lookupRows.map(this._createTreeNode.bind(this));
```
Typos in commit message:
![grafik](https://github.com/user-attachments/assets/8e8d4c83-c77f-4d18-9fc7-6b785332a28c)

cguglielmo replied:
This commit should not be in this PR.... :/

bschwarzent commented on eclipse-scout-core/src/resizable/Resizable.ts:
```diff
@@ -229,15 +231,42 @@ export class Resizable implements ResizableModel, ObjectWithType {
+  /**
+   * Finishes the resizing by cleaning up all temporary states and triggering the `resizeEnd` event with the initial and new bounds.
+   */
+  public finish() {
```
We usually don't write the "public" modifier in TS.

cguglielmo replied:
true

bschwarzent commented on eclipse-scout-core/src/main.less:
```diff
@@ -326,18 +326,16 @@ button {
-.row-resize * { /* NOSONAR (universal selector is okay here) */
-  cursor: row-resize !important; /* NOSONAR (!important is okay here) */
-}
+#scout.resize(col);
```
Maybe name the mixin "resize-cursor" instead of just "resize"? Right now, it is not entirely obvious what this block does (without looking at the documentation of the mixin).

cguglielmo replied:
good idea

bschwarzent commented on eclipse-scout-core/src/style/mixins.less:
```diff
@@ -633,4 +633,14 @@
+  .resize(@edge) {
+    .@{edge}-resize,
+    .@{edge}-resize * { /* NOSONAR (universal selector is okay here) */
+      cursor: ~'@{edge}-resize !important'; /* NOSONAR !important is necessary to override any defined cursor on underlying components */
```
You probably tested this, but why is the value a string? Isn't this usually done to prevent LESS from converting anything inside? But @{edge} should be replaced with the actual value.

cguglielmo replied:
This is the only way I got it to compile, the output looks correct, the single quotes will be gone and the variable replaced.

cguglielmo committed:
Resizable: don't use the same event name as the browser

There is a native 'resize' event triggered by the browser when the
window is resized.
Because the resizable element triggers an event with the same name on a
DOM element, it bubbles up to the window executing the event handlers
listening for window resizes, which is unexpected.
The event handlers could check the target, but since this is an
artificial event and not the same as the one sent by the browser, the
handlers should not need to take care of that.

cguglielmo committed:
Resizable/MoveSupport: only allow resizing with the left mouse button

The right mouse button typically opens the context menu.
If resizing can be started with that button, the context menu would open
as well, which is confusing.

Also ensure the resize overlay won't stay in the DOM, e.g. if the right
button is pressed while resizing has already started, or mouse is
released outside the window without triggering a mouseup event.

389402
